vermouth/tests/test_ff_files.py

```diff
@@ -0,0 +1,1019 @@
+            ),
+            {'bonds': [
+                vermouth.molecule.Interaction(
+                    atoms=['BB', '--SC1'],
+                    parameters=['1', '0.3', '100'],
+                    meta={},
+                ),
+            ]},
+            (('BB', '--SC1'),),
+        ),
+        (  # Base case with implicit order in the attributes
+            """
+            [link]
+            [bonds]
+            BB SC1 {"order": ">>"} 1 0.3 100
+            """,
+            (
+                ('BB', {'atomname': 'BB', 'order': 0}),
+                ('>>SC1', {'atomname': 'SC1', 'order': '>>'}),
+            ),
+            {'bonds': [
+                vermouth.molecule.Interaction(
+                    atoms=['BB', '>>SC1'],
+                    parameters=['1', '0.3', '100'],
+                    meta={},
+                ),
+            ]},
+            (('BB', '>>SC1'),),
+        ),
+        (  # Inhibit edges
+            """
+            [link]
+            [bonds]
+            BB SC1 1 0.3 100 {"edge": false}
+            """,
+            (
+                ('BB', {'atomname': 'BB', 'order': 0}),
+                ('SC1', {'atomname': 'SC1', 'order': 0}),
+            ),
+            {'bonds': [
+                vermouth.molecule.Interaction(
+                    atoms=['BB', 'SC1'],
+                    parameters=['1', '0.3', '100'],
+                    meta={'edge': False},
+                ),
+            ]},
+            (),
+        ),
+        (  # Unknown interaction that does not add an edge
+            """
+            [link]
+            [unknown_interaction]
+            BB SC1 SC2
+            """,
+            (
+                ('BB', {'atomname': 'BB', 'order': 0}),
+                ('SC1', {'atomname': 'SC1', 'order': 0}),
+                ('SC2', {'atomname': 'SC2', 'order': 0}),
+            ),
+            {'unknown_interaction': [
+                vermouth.molecule.Interaction(
+                    atoms=['BB', 'SC1', 'SC2'],
+                    parameters=[],
+                    meta={},
+                ),
+            ]},
+            (),
+        ),
+        (  # Atom already in the link
+            """
+            [link]
+            [atoms]
+            BB {"element": "Q"}
+            [bonds]
+            BB SC1 1 0.3 100
+            """,
+            (
+                ('BB', {'atomname': 'BB', 'order': 0, 'element': 'Q'}),
+                ('SC1', {'atomname': 'SC1', 'order': 0}),
+            ),
+            {'bonds': [
+                vermouth.molecule.Interaction(
+                    atoms=['BB', 'SC1'],
+                    parameters=['1', '0.3', '100'],
+                    meta={},
+                ),
+            ]},
+            (('BB', 'SC1'),),
+        ),
+        (  # Interaction atoms have attributes
+            """
+            [link]
+            [bonds]
+            BB {"custom": 1, "plop": "toto"} SC1 {"toto": "tata"} 1 0.3 100
```

Comment:
Is this desirable? Isn't it better to force a separate [atoms] section in these cases?

Reply:
It would make writing links super über really completely unreasonably tedious. I know: I tried.